### ISSUE TITLE
reddit-settings: Patching error after changing options.json value to `true`

### ISSUE THREAD

**Issue by KobeW50** [inotia00/ReVanced_Extended#1067]
### Type

Error while patching

### Tools used

Revancify

### Application

Reddit v2023.26.0

### Bug description

After changing the reddit-settings field in the patch options from `false` to `true`, there will be an error while patching and it will abort.

I tried with the recommended patches (all), and also tried excluding the reddit-settings patch, and even that didn't work.

Interestingly, even after changing the patch option back to `false` patching won't work (even for YT, and I assume YTM as well) until I delete the options.json file and generate a new one. I am unsure if this is a Revancify issue or an RVX issue.

### Steps to reproduce

1. Generate a options.json file.
2. Edit the reddit-settings value to `true`. Then save.
3. Patch Reddit 2023.26.0 with the default patches.

### Relevant log output

```shell
java.lang.ClassCastException: class java.lang.Boolean cannot be cast to class java.lang.String (java.lang.Boolean and java.lang.String are in module java.base of loader 'bootstrap')
	at app.revanced.patcher.patch.PatchOption$StringOption$1.invoke(PatchOption.kt:156)
	at app.revanced.patcher.patch.PatchOption.setValue(PatchOption.kt:108)
	at app.revanced.utils.Options.setOptions(Options.kt:145)
	at app.revanced.utils.Options.setOptions(Options.kt:86)
	at app.revanced.cli.command.MainCommand.run(MainCommand.kt:152)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at app.revanced.cli.main.MainKt.main(Main.kt:7)



Variant: nonRoot
Arch: arm64-v8a
App: Reddit v2023.26.0
CLI: inotia00-cli-v2.21.6.jar
Patches: inotia00-patches-v2.181.4.jar
Integrations: inotia00-integrations-v0.111.4.apk
Patches argument: -i disable-screenshot-popup
-i hide-ads
-i hide-chat-button
-i hide-create-button
-i hide-discover-button
-i open-links-directly
-i open-links-externally
-i premium-icon-reddit
-i reddit-settings
-i sanitize-sharing-links
```

### Screenshots or videos

In the screen recording I patch 3 times. The first is patching with a clean options.json, which works. The second is after changing the value to `true`. The third is after changing the value back to `false`.

https://github.com/inotia00/ReVanced_Extended/assets/84587632/ec19a6d2-043e-4d85-bdd7-f1be5aa701d3

### Additional context

Termux v0.118.0

Importantly, I have a feeling I am misunderstanding the value that should be entered in the options.json. It is implied from what you wrote in the release notes that there are multiple values, instead of just `true`. 

"preferences are added to settings only when options.json values are specified
(So when patching with ReVanced Manager, no switches are added to the settings. for compatibility with ReVanced Manager)"

### Device Environment

Samsung. Android 13

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.
- [X] I have written the title and contents in English.
- [X] This issue does not reproduce on unpatched YouTube or YT Music.

**Comment by KobeW50** [comment]
@leagueofperez
It seems like from your issue (#1068), when you patched with the CLI after changing the value to `true`, you did not get an error. Is this correct?

**Comment by leagueofperez** [comment]
> @leagueofperez It seems like from your issue (#1068), when you patched with the CLI after changing the value to `true`, you did not get an error. Is this correct?

I've changed `AddRedditSettings`from `false`  to `true` and didn't got any error.

**Comment by inotia00** [comment]
completely remove the options.json file and try again

**Comment by KobeW50** [comment]
> completely remove the options.json file and try again

So I should delete it, then patch any app (without generating a new options.json before patching), and then edit the options.json and patch reddit afterwards?

**Comment by KobeW50** [comment]
> completely remove the options.json file and try again

These steps did not work unfortunately.

1. delete the options.json
2. (Without generating a new options.json
beforehand) Patched 2023.23.0
3. Edited the reddit-settings value to true
4. Patched 2023.25.1

The error was the same

**Comment by inotia00** [comment]
For some reason, editing `options.json` with revancify's editor seems to save "true" as a boolean and not as a string.

**Comment by KobeW50** [comment]
Ohhhh that explains it

**Comment by inotia00** [comment]
This issue has been fixed in v2.181.6
